packages/notte-browser/src/notte_browser/controller.py

```diff
@@ -86,6 +87,8 @@ async def execute_browser_action(self, window: BrowserWindow, action: BaseAction
                     await window.page.keyboard.press("PageDown")
             case ScrapeAction():
                 raise NotImplementedError("Scrape action is not supported in the browser controller")
+            case ReadPDFAction():
```

Comment:
I would remove this action and put into the scraping behaviour (cf other comments)